src/api/IServer.hh

```diff
@@ -16,11 +16,12 @@ public:
     virtual const char* Name() const noexcept = 0;
 
     // === user functions ===
-
-    virtual Login::Response Login(const Login::Request& request) noexcept = 0;
-    virtual Logout::Response Logout(const Logout::Request& request) noexcept = 0;
-    virtual StartSession::Response StartSession(const StartSession::Request& request) noexcept = 0;
-    virtual Ping::Response Ping(const Ping::Request& request) noexcept = 0;
+    // TBD: Overrides can throw
+    [[gsl::suppress(f.6)]] virtual Login::Response Login(const Login::Request& request)    = 0;
+    [[gsl::suppress(f.6)]] virtual Logout::Response Logout(const Logout::Request& request) = 0;
+    [[gsl::suppress(f.6)]] 
+    virtual StartSession::Response StartSession(const StartSession::Request& request) = 0;
+    [[gsl::suppress(f.6)]] virtual Ping::Response Ping(const Ping::Request& request) = 0;
```

Comment:
None of the toolkit code is designed to handle exceptions very well, so I'd rather have everything written assuming `noexcept` and patch the areas that can throw exceptions with try/catch blocks.

Reply:
Alright, I'll suppress where it can throw but mark as noexcept